Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs

```diff
@@ -5,35 +5,141 @@
 namespace Notatnik_parser.Notepad
 {
     public class NotepadMenu
-    {           
-        private void SaveFile(string textNoteParameter)
-        {            
-            string fileName = "LastNote.txt";
+    {    
+        string pathInstructions = "Podaj pełną ścieżkę z nazwą pliku lub samą nazwę pliku z rozszerzeniem, żeby {0} domyślnego folderu:";
+        
+        public void Run() {
 
-            using (FileStream fs = File.Create(fileName))
-                {
-                    Byte[] textNote = new UTF8Encoding(true).GetBytes(textNoteParameter);
-                    fs.Write(textNote, 0, textNote.Length);
-                }            
+            ShowNotpadMenu();
+            MenageUserInput();
         }
 
-        public void Run()
-        {   
-            string note;
+        public void ShowNotpadMenu()
+        {
+            Console.WriteLine("Witamy w notatniku. Wybierz:");
+            Console.WriteLine("1 - nowa notatka");
+            Console.WriteLine("2 - otwórz notatkę");
+        }
 
-            Console.WriteLine("Witamy w notatniku. Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+        private void MenageUserInput()
+        {
+            string insertValue;
+            do
+            {
+                insertValue = Console.ReadLine();
 
-            note = Console.ReadLine();
+            } while (!IsUserInputValid(insertValue));
 
-            ConsoleKeyInfo typedKey = Console.ReadKey();
-            if (typedKey.Key == ConsoleKey.Enter)
+            GoToUserChoice(insertValue);
+        }
+
+        private bool IsUserInputValid(string userInput)
+        {
+            if (userInput == "1" || userInput == "2")
+            {
+                return true;
+            }
+
+            Console.WriteLine("Proszę wybrać numer z listy");
+            return false;
+        }
+
+        private void GoToUserChoice(string userChoice)
+        {
+            switch (userChoice)
             {
-                Console.WriteLine("To jest twoja notatka, która została zapisana do pliku txt: " + note);
-                SaveFile(note);
-                
+                case "1":
+                    WriteNewNote();
+                    break;
+                case "2":
+                    MenageUserPathInput();
+                    break;
+                default:
+                    Console.WriteLine("Błąd");
+                    break;
+            }
+        }
+
+        private void WriteNewNote()
+        {       
+            Console.WriteLine("Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+            string note = Console.ReadLine();
+            ConsoleKeyInfo typedKey = Console.ReadKey();
+
+            if (typedKey.Key == ConsoleKey.Enter && SaveFile(note))
```

Comment:
saveFile shoudn't execute in if. You should execude saveFile method and then pass returned value to if statement. You can try at first ask about new path and check it and then ask about data for file. Now save method method saves the file and asks about path. It's two responsibilities